pkg/resource/deploy/step.go

```diff
@@ -840,7 +858,28 @@ func (s *RefreshStep) Apply(preview bool) (resource.Status, StepCompleteFunc, er
 		s.new = resource.NewState(s.old.Type, s.old.URN, s.old.Custom, s.old.Delete, resourceID, inputs, outputs,
 			s.old.Parent, s.old.Protect, s.old.External, s.old.Dependencies, initErrors, s.old.Provider,
 			s.old.PropertyDependencies, s.old.PendingReplacement, s.old.AdditionalSecretOutputs, s.old.Aliases,
-			&s.old.CustomTimeouts, s.old.ImportID, s.old.RetainOnDelete, s.old.DeletedWith)
+			&s.old.CustomTimeouts, s.old.ImportID, s.old.RetainOnDelete, s.old.DeletedWith, s.old.Created, s.old.Modified)
+		complete = func() {
+			hasStateChange := func(refreshed plugin.ReadResult, inputs resource.PropertyMap, outputs resource.PropertyMap) bool {
+				inputsChange := false
+				if s.old != nil {
+					inputsChange = !refreshed.Inputs.DeepEquals(s.old.Inputs)
+				}
+
+				outputsChange := false
+				if s.old != nil {
```

Comment:
The if condition doesn't change between the two cases.
We could combine these blocks.

```go
var inputsChange, outputsChange bool
if s.old != nil {
	inputsChange = !refreshed.Inputs.DeepEquals(s.old.Inputs)
	outputsChange = !refreshed.Outputs.DeepEquals(s.old.Outputs)
}
```

Reply:
Pushed a fix.